src/SecTester.Scan/Internal/SemaphoreLock.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SecTester.Scan.Internal;
+
+internal class SemaphoreLock : IDisposable
```

Comment:
Your implementation is not safe, see #75

Reply:
This is a question to you, no to me. It was designed to be used internally in the repeater project, if you need to do the same, feel free to extract it into the core project.